src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java

```diff
@@ -178,12 +188,12 @@ private static DetailAST getParent(DetailAST commentBlock) {
 
         while (true) {
             final int type = result.getType();
-            if (type == TokenTypes.TYPE || type == TokenTypes.MODIFIERS
-                    || type == TokenTypes.ANNOTATION || type == TokenTypes.ANNOTATIONS
-                    || type == TokenTypes.ARRAY_DECLARATOR || type == TokenTypes.TYPE_PARAMETERS
-                    || type == TokenTypes.DOT) {
+            if (TYPES_ARRAY_LIST.contains(type)) {
```

Comment:
I might miss something, this update is just code refactoring, and not related to functional fix. If so , please move to searate comit of the same PR, commit with prefix "minor".

Reply:
This was done to reduce cyclomatic complexity (11) reported by `mvn clean verify`.

Reply:
So it is unrelated to functional fix and fortunately not mixed with other update, please move it to separate commit with message `minor: reduce cyclomatic complexity before next update`

Reply:
Fixed.